proposed/2023/nuget-package-search.md

```diff
@@ -0,0 +1,74 @@
+# NuGet Package Search
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- [Nigusu](https://github.com/Nigusu-Allehu) <!-- GitHub username link -->
+- [GitHub Issue](https://github.com/NuGet/Home/issues/6060) <!-- GitHub Issue link -->
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+This specification will discuss three key improvements to the NuGet package search functionality. These improvements are: adding an `-ExactMatch` option to the `search` command, deprecating the `list` command in favor of `search`, and introducing a `dotnet nuget search` command to the Dotnet CLI.
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Improving the nuget search functionality across all platforms is highly requested by customers. As shown in this [issue](https://github.com/NuGet/Home/issues/5138), customers would like to have `-ExactMatch` option added to the search command. In this other [issue](https://github.com/NuGet/Home/issues/6060), we can see that it has been repeatedly requested by customers for the Dotnet CLI to adopt a search functionality. Lastly, as discussed in this [issue](https://github.com/NuGet/Home/issues/7912) the `nuget list` command is redundant with `nuget search` as both commands basically display a list of packages from a source. The only difference between the two is that the search command allows for filters. In addition, the `nuget list` command can be confusing as there is a `nuget list` command in the Dotnet CLI with a different functionality: lists configured sources and client certificates. Addressing these issues will provide users with a more straight-forward search functionality on both the Dotnet CLI and nuget.exe.
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+#### 1. Adding `-ExactMatch` Option: 
```

Comment:
Does the `dotnet nuget search` command output include information about `deprecated and/or vulnerable` package versions? If we consider this command as equivalent to the NuGet.org search or Visual Studio Package Manager UI search, then I think it would be beneficial to customers to display such information.

Reply:
+1 on this question. NuGet.org search endpoint now contains both vulnerable and deprecated information so you could consider leveraging that.

Reply:
Thank you @kartheekp-ms , here is a separate [spec ](https://github.com/NuGet/Home/pull/12886) for dotnet package search. As you recommended, I have added `deprecated and/or vulnerable` as part of the output.